nrfx/drivers/src/nrfx_uarte.c

```diff
@@ -1637,8 +1639,16 @@ nrfx_err_t nrfx_uarte_rx_ready(nrfx_uarte_t const * p_instance, size_t * p_rx_am
         return NRFX_ERROR_FORBIDDEN;
     }
 
-    if (nrfy_uarte_event_check(p_instance->p_reg, NRF_UARTE_EVENT_ENDRX) ||
-        !nrfy_uarte_enable_check(p_instance->p_reg))
+    if (!nrfy_uarte_enable_check(p_instance->p_reg))
+    {
+        if (p_rx_amount)
+        {
+            *p_rx_amount = nrfy_uarte_rx_amount_get(p_instance->p_reg);
+        }
+
+        return NRFX_ERROR_ALREADY;
```

Comment:
This return value should be documented.
I'm not sure what does this value actually mean, is it something that may normally occur (and that's why the amount is also returned)? Or is it an actual error and maybe something like `NRFX_ERROR_INVALID_STATE` would be more appropriate?

Reply:
It may occur when uarte is suspended by PM. In that case we fail to read RX in a blocking way.